lib/features/base/base_mailbox_controller.dart

```diff
@@ -554,4 +601,23 @@ abstract class BaseMailboxController extends BaseController {
     mailboxNode = personalMailboxTree.value.findNodeOnFirstLevel((node) => node.item.name?.name.toLowerCase() == name);
     return mailboxNode;
   }
+
+  void copySubaddressAction(BuildContext context, String subaddress) {
+    Clipboard.setData(ClipboardData(text: subaddress));
+    appToast.showToastSuccessMessage(context, AppLocalizations.of(context).emailSubaddressCopiedToClipboard);
+  }
+
+  String getSubaddress(String userEmail, String folderName) {
+    if (folderName.isEmpty) {
+      throw const FormatException('folder name should not be empty');
+    }
+    int atIndex = userEmail.indexOf('@');
+    String localPart = userEmail.substring(0, atIndex);
+    String domain = userEmail.substring(atIndex + 1);
+    if (atIndex > 0 && localPart.isNotEmpty && domain.isNotEmpty) {
+      return "$localPart+$folderName@$domain";
+    } else {
+      throw const FormatException('Invalid email format');
+    }
+  }
```

Comment:
To improve performance should use:

```suggestion
String getSubaddress(String userEmail, String folderName) {
  if (folderName.isEmpty) {
    throw FormatException('Folder name should not be empty');
  }

  final atIndex = userEmail.indexOf('@');
  if (atIndex <= 0 || atIndex == userEmail.length - 1) {
    throw FormatException('Invalid email format');
  }

  return '${userEmail.substring(0, atIndex)}+$folderName@${userEmail.substring(atIndex + 1)}';
}
```